### PR TITLE
Added pre-commit hooks and refactored script to generate. Fixes #6

### LINKED ISSUE #6: Add pre-commit hooks
Following the conversation in [PR 4](https://github.com/ebi-ait/ingest-graph-validator/pull/4), I would like to add a pre-commit hook that would run the script `.readme/update_test_index.py` prior to any commit to the repo.

This would ensure an updated list of all the tests currently under the folder `graph_test_set`

### PR DESCRIPTION
**Pull request by ESapenaVentura**
**Changelog**:
- Added pre-commit hook to run when adoc files are modified (Due to limitations of pre-commit package detecting deleted files, it always runs the `.sh` script but the python script only when there are modified `adoc` files)
- Added `.readme/TESTS_README.template` template to create README.md file with script
- Refactored and modified `.readme/update_test_index.py` script to use the template

### CODE REVIEW

**Comment by ESapenaVentura** [comment]
No, it wasn't intentional, just a byproduct of using pycharm

I'll delete them immediately and add to the gitignore. Thanks for noticing!

**Comment by mshadbolt** [comment]
I also think that the `how to run the tests` section in the tests readme should actually go into the main repo readme

File: graph_test_set/README.md [hunk 1]
```diff
@@ -1,11 +1,73 @@
+# Tests directory
+In this folder you will find the tests the Human Cell Atlas uses for validating entities in their experiments
+
+## How to run the tests
```
Comment (mshadbolt) [inline graph_test_set/README.md:4]: This section should go in the main repo readme instead

**Comment by ESapenaVentura** [comment]
Made the changes - Asking for another review!

**Comment by mshadbolt** [comment]
Sorry I wasn't totally clear.

I meant to leave the 'how to add new tests' section in the /graph_test_set README.md and add the 'how to run the tests' in the main repo README.md

Also would we need to add it to the `README.template` in `.readme` rather than directly to the README.md file in the root directory?

**Comment by ESapenaVentura** [comment]
Part by part:
>I meant to leave the 'how to add new tests' section in the /graph_test_set README.md and add the 'how to run the tests' in the main repo README.md

Ok, I understand better now. Yeah, it does make sense to have those 2 divided

>Also would we need to add it to the README.template in .readme rather than directly to the README.md file in the root directory?

Probably both for now as we don't have any process for running the `update_subs_id` script? but good catch on that I didn't add it to the template

**Comment by mshadbolt** [comment]
ok yep I understand. everything else looks good i think

**Comment by ESapenaVentura** [comment]
Great! I have made the last changes and I'll proceed to merge :)

**Pull request merged**